openvdb/tools/LevelSetMeasure.h

```diff
@@ -57,67 +58,62 @@ namespace tools {
 template<class GridType>
 inline Real
 levelSetArea(const GridType& grid, bool useWorldSpace = true);
 
 /// @brief Return the volume of a narrow-band level set surface.
 ///
 /// @param grid          a scalar, floating-point grid with one or more disjoint,
-///                      closed isosurfaces at the given @a isovalue
+///                      closed level set surfaces
 /// @param useWorldSpace if true the volume is computed in
 ///                      world space units, else in voxel units.
 ///
-/// @throw TypeError if @a grid is not scalar or not floating-point or not a level set.
+/// @throw TypeError if @a grid is not scalar or not floating-point or not a level set or empty.
 template<class GridType>
 inline Real
 levelSetVolume(const GridType& grid, bool useWorldSpace = true);
 
-/// @brief Compute the surface area and volume of a narrow-band level set.
+/// @brief Return the Euler Characteristics of a narrow-band level set surface (possibly disconnected).
 ///
 /// @param grid          a scalar, floating-point grid with one or more disjoint,
-///                      closed isosurfaces at the given @a isovalue
-/// @param area          surface area of the level set
-/// @param volume        volume of the level set surface
-/// @param useWorldSpace if true the area and volume are computed in
-///                      world space units, else in voxel units.
+///                      closed level set surfaces
 ///
-/// @throw TypeError if @a grid is not scalar or not floating-point or not a level set.
+/// @throw TypeError if @a grid is not scalar or not floating-point or not a level set or empty.
 template<class GridType>
-inline void
-levelSetMeasure(const GridType& grid, Real& area, Real& volume, bool useWorldSpace = true);
+inline int
+levelSetEulerCharacteristic(const GridType& grid);
 
-/// @brief Compute the surface area and volume of a narrow-band level set.
+/// @brief Return the genus of a narrow-band level set surface.
 ///
 /// @param grid          a scalar, floating-point grid with one or more disjoint,
-///                      closed isosurfaces at the given @a isovalue
-/// @param area          surface area of the level set
-/// @param volume        volume of the level set surface
-/// @param avgCurvature  average mean curvature of the level set surface
-/// @param useWorldSpace if true the area, volume and curvature are computed in
-///                      world space units, else in voxel units.
+///                      closed level set surfaces
+/// @warning The genus is only well defined for a single connected surface
 ///
-/// @throw TypeError if @a grid is not scalar or not floating-point or not a level set.
+/// @throw TypeError if @a grid is not scalar or not floating-point or not a level set or empty.
 template<class GridType>
-inline void
-levelSetMeasure(const GridType& grid, Real& area, Real& volume, Real& avgCurvature,
-                bool useWorldSpace = true);
```

Comment:
As discussed, if we can bring these back, mark them as deprecated (perhaps move to the end of the file?) and alias them to the new methods that would be good. Ideally we'd do this in the LevelSetMeasure class too, but I'm unclear on how much work that will be

Reply:
I will add this function back but deprecate it